src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/innerassignment/InputInnerAssignment.java

```diff
@@ -111,4 +111,18 @@ public ChildParent getParent() {
             return this;
         }
     }
+
+    void method() throws IOException {
+        Integer line;
+        FileInputStream file = null;
+        while (!((line = file.read()) != null)) {}
+        while ((line = file.read()) != null && line < 3) {}
+        while ((line = file.read()) != null && line < 3 && line > 5) {}
+        while ((line = file.read()) != null || line < 3) {}
+        while ((line = file.read()) != null || line < 3 || line > 5) {}
+        while ((line = file.read()) != null & line < 3) {}
+        while ((line = file.read()) != null & line < 3 & line > 5) {}
+        while ((line = file.read()) != null | line < 3) {}
+        while ((line = file.read()) != null | line < 3 | line > 5) {}
```

Comment:
After seeing https://strkkk.github.io/6805/RxJava/xref/checkstyle/contribution/checkstyle-tester/repositories/RxJava/src/main/java/rx/internal/operators/OperatorMergeMaxConcurrent.java.html#L264 , please add an example where assignment isn't first thing in the condition.

Reply:
Done